File: tests/src/main/java/com/github/klee0kai/test/di/base_phone/qualifiers/DataStorageSize.java
package com.github.klee0kai.test.di.base_phone.qualifiers;

import java.util.Objects;

public class DataStorageSize {

    public String size;

    public DataStorageSize(String size) {
        this.size = size;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        DataStorageSize that = (DataStorageSize) o;
        return Objects.equals(size, that.size);
    }

    @Override
    public int hashCode() {
        return Objects.hash(size);
    }
}

File: test_kotlin/src/main/kotlin/com/github/klee0kai/test/lifecycle/di/qualifier/DataStorageSize.kt
package com.github.klee0kai.test.lifecycle.di.qualifier

import java.util.*

class DataStorageSize(var size: String) {
    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as DataStorageSize
        return size == that.size
    }

    override fun hashCode(): Int {
        return Objects.hash(size)
    }
}